lib/puppet/provider/user/hpux.rb

```diff
@@ -63,4 +75,43 @@ def password
       return ent.passwd
     end
   end
+
+  def trusted
+      trusted_sys = %x(/usr/lbin/getprpw root 2>&1)
+      if trusted_sys.chomp == "System is not trusted."
+         "NotTrusted"
+      else
+         "Trusted"
+      end
+  end
+
+  def trust2
```

Comment:
The trusted returns a string, the trust2 returns a boolean.

This is all used for the mocks / rspec code, not the actual functional 
coding required to make it work.

Again I have not found any good references on writing rspec 
specifications to test this code or proper mocking, so I have had to 
hack the code to make the testing properly return as I expected.

Any suggestions to references would be greatly appreciated.

On 1/2/2015 2:07 PM, Michael Smith wrote:
>
> In lib/puppet/provider/user/hpux.rb 
> <https://github.com/puppetlabs/puppet/pull/3443#discussion-diff-22419033>:
>
> > @@ -63,4 +75,43 @@ def password
> >        return ent.passwd
> >      end
> >    end
> > +
> > +  def trusted
> > +      trusted_sys = %x(/usr/lbin/getprpw root 2>&1)
> > +      if trusted_sys.chomp == "System is not trusted."
> > +         "NotTrusted"
> > +      else
> > +         "Trusted"
> > +      end
> > +  end
> > +
> > +  def trust2
>
> This appears to be nearly identical to #trusted, and unused. Why is it 
> included?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/puppetlabs/puppet/pull/3443/files#r22419033>.
>